src/protocol.rs

```diff
@@ -40,11 +41,28 @@ pub struct Field {
     pub value: FieldData,
 }
 
+/// Container struct for timestamp attributes.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Timestamp {
+    pub value: u64,
+}
+
 impl LineProtocol {
-    pub fn new(measurement: String, tags: Option<String>, fields: String) -> Self {
+    pub fn new(
+        measurement: String,
+        tags: Option<String>,
+        fields: String,
+        timestamp: Option<String>,
+    ) -> Self {
         match tags {
-            Some(t) => Self(format!("{},{} {}\n", measurement, t, fields)),
-            None => Self(format!("{} {}\n", measurement, fields)),
+            Some(t) => match timestamp {
+                Some(ts) => Self(format!("{},{} {} {}\n", measurement, t, fields, ts)),
+                None => Self(format!("{},{} {}\n", measurement, t, fields)),
+            },
+            None => match timestamp {
+                Some(ts) => Self(format!("{} {} {}\n", measurement, fields, ts)),
+                None => Self(format!("{} {}\n", measurement, fields)),
+            },
         }
```

Comment:
This seems clunky/repeated code. Perhaps something like

```rust
let mut lp = match tags {
    Some(t) => Self(format!("{},{} {}", measurement, t, fields)),
    None => Self(format!("{} {}", measurement, fields)),
}

if let Some(ts) = timestamp {
    lp.push_str(format!(" {}", ts));
}

lp.push_str("/n");
Self(lp)
```

Rough idea. Less repetition would be nice I think.

Reply:
Much better, thank you.